src/main/java/io/aiven/kafka/connect/opensearch/OpensearchBasicAuthConfigurator.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 Aiven Oy
+ * Copyright 2016 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.opensearch;
+
+import io.aiven.kafka.connect.opensearch.spi.OpensearchClientConfigurator;
+
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+
+/**
+ * Adds basic authentication to the {@index HttpAsyncClientBuilder} for Opensearch client 
+ * if configured.
+ */
+public class OpensearchBasicAuthConfigurator implements OpensearchClientConfigurator {
+    @Override
+    public boolean appliesTo(final OpensearchSinkConnectorConfig config) {
+        return config.isAuthenticatedConnection();
```

Comment:
Nit: just for the sake of making this a helpful reference implementation for others, do you think we could define all the basic auth-related properties in this class instead of in the `OpensearchSinkConnectorConfig`? Other developers implementing custom configurators won't be able to modify that class to add methods like `isAuthenticatedConnection` or `connectionUsername`.